server/cloudformation/ecs.go

```diff
@@ -172,39 +190,13 @@ func (p *ECSServiceResource) delete(ctx context.Context, service, cluster *strin
 	return nil
 }
 
-// Certain parameters cannot be updated on existing services, so we need to
-// create a new physical resource.
-func requiresReplacement(new, old *ECSServiceProperties) bool {
-	eq := reflect.DeepEqual
-
-	if !eq(new.Cluster, old.Cluster) {
-		return true
-	}
-
-	if !eq(new.Role, old.Role) {
-		return true
-	}
-
-	if !eq(new.ServiceName, old.ServiceName) {
-		return true
-	}
-
-	if !eq(new.LoadBalancers, old.LoadBalancers) {
-		return true
-	}
-
-	return false
-}
-
-var letters = []rune("ABCDEFGHIJKLMNOPQRSTUVWXYZ123456789")
-
-// Generates a random 12 character string (similar to how standard
-// CloudFormation works).
-func postfix() string {
-	n := 12
-	b := make([]rune, n)
-	for i := range b {
-		b[i] = letters[rand.Intn(len(letters))]
+// It's important that the postfix we append to the service name
+// is deterministic based on the non updateable fields of the
+// service, otherwise ClientToken has no effect on idempotency.
+func postfix(p *ECSServiceProperties) (string, error) {
+	h, err := p.Hash()
```

Comment:
Thinking about this more, hashing the request id would be better. If we hash the props, you could potentially get into a scenario where a service gets re-created with the same name as a service that was previously deleted. We've seen a lot of weird behavior when that happens (complaining about the load balancer not existing, saying it's in a draining state, etc), so best to avoid it.